crates/router/src/db/payment_method.rs

```diff
@@ -105,44 +106,122 @@ impl PaymentMethodInterface for Store {
 impl PaymentMethodInterface for MockDb {
     async fn find_payment_method(
         &self,
-        _payment_method_id: &str,
+        payment_method_id: &str,
     ) -> CustomResult<storage::PaymentMethod, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let payment_methods = self.payment_methods.lock().await;
+        let payment_method = payment_methods
+            .iter()
+            .find(|pm| pm.payment_method_id == payment_method_id)
+            .cloned();
+
+        match payment_method {
+            Some(pm) => Ok(pm),
+            None => Err(errors::StorageError::ValueNotFound(
```

Comment:
We can instead use this:
```suggestion
            None => Err(errors::ConnectorError::PaymentMethodNotFound())
```

Reply:
Could it be that it is actually: `errors::ApiErrorResponse::PaymentMethodNotFound()` ?

Reply:
Unless you mean to add a new error

Reply:
We're required to return `StorageError` from `MockDb` implementations, so it's fine to return `StorageError::ValueNotFound`.

Resolving this conversation and the others.